scripts/check-markdown-links.ts

```diff
@@ -0,0 +1,189 @@
+  const internalLink = `<td data-title="${tableHeaders[1]}"><a href="${file}" target="_blank">${filePath}</a></td>`;
+  const brokenLink = `<td data-title="${tableHeaders[2]}"><a href="${url}" target="_blank">${url}</a></td>`;
+  const priority = `<td data-title="${tableHeaders[3]}">${
+    status === 404
+      ? `<strong>High:</strong> 404 broken link.`
+      : `<strong>Investigate status:</strong> ${status}`
+  }</td>`;
+  const content = `<tr> ${externalLink + internalLink + brokenLink + priority}</tr>`;
+  return content;
+};
+
+// Prevents us trying to call localhost urls.
+const isValidUrl = (url: string): boolean => {
+  let urlHolder: URL;
+  try {
+    urlHolder = new URL(url);
+  } catch (error) {
+    console.log(
+      `${url} could not be turned into a proper URL. Maybe explicitly exclude this type of thing?`
+    );
+    return false;
+  }
+  if (urlHolder.hostname.includes('localhost')) {
+    return false;
+  }
+  return true;
+};
+
+// Wrapper function for Synchronous file writing.
+const writeToFile = (
+  outputFileName: string,
+  content = '',
+  flag: WriteFileOptions,
+  errorMessage = 'Error!',
+  successMessage?: string
+) => {
+  const baseCSS = `<link rel="stylesheet" href="https://design.cms.gov/cdn/design-system/10.1.1/css/index.css" />`;
+  const coreCSS = `<link rel="stylesheet" href="https://design.cms.gov/cdn/design-system/10.1.1/css/core-theme.css" />`;
+  const htmlOutput = `<!DOCTYPE html><html><head>${baseCSS}${coreCSS}</head><body>${content}</body></html>`;
+  try {
+    fs.writeFileSync(outputFileName, htmlOutput, flag);
+    successMessage && console.log(successMessage);
+  } catch (err) {
+    console.error(errorMessage.concat(` ${err}`));
+  }
+};
+
+// Extract links from a particular md/mdx file and place into predefined object:
+const findLinks = (filePath: string, regex: RegExp): void => {
+  try {
+    const fileContent = fs.readFileSync(filePath, 'utf8');
+    const matches: RegExpMatchArray | null = fileContent.match(regex);
+    if (matches) {
+      (urlsToCheck as FilesAndLinks)[filePath] = matches.filter((match) => {
+        if (isValidUrl(match)) {
+          return match;
+        }
+      });
+    }
+  } catch (error) {
+    console.log(
+      `An error occurred when parsing this file: ${filePath}. \nHere's the error:\n ${error}`
+    );
+  }
+};
+
+// Use node-fetch to query urls and return a status code:
+const curl = async (file: string, url: string, lastItem: boolean): Promise<void> => {
+  const spliceFrom = file.split('/').indexOf('design-system');
+  const filePath: string = file.split('/').splice(spliceFrom).join('/');
+  const content = '';
+  fetch(url).then((res) => {
+    if (res.status !== 200) {
+      const rowContent = tableBodyContent(file, filePath, res.status, url);
+      tableBodyContainer.push(rowContent);
+    }
+    if (lastItem) {
+      const outputTable = table(tableHeader(), tableBodyContainer.join(''));
+      writeToFile(
```

Comment:
It might be nice to add a success message like "Broken links found! Run `open brokenLinkReport.html` to view the list."